src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -4750,19 +4750,30 @@ public static void enableAutoCompaction() {
         }
     }
 
+    public void internalDisableNode() {
+        logger.info("Stopping transports and disabling auto compaction");
+        instance.stopTransports();
+        disableAutoCompaction();
+    }
+
+    @Override
+    public void disableNode() {
+        setMode(Mode.DISABLED, "Node has been disabled", true);
```

Comment:
i think we should only be allowed to disable a node that is either already disabled or normal.  otherwise we might accidentally wipe important information about the status